src/main/scala/eventstore/PersistentSubscriptionActor.scala

```diff
@@ -0,0 +1,211 @@
+package eventstore
+import akka.actor.Status.Failure
+import akka.actor.{ ActorRef, Props }
+import eventstore.PersistentSubscription.{ Ack, EventAppeared }
+import eventstore.ReadDirection.Forward
+import eventstore.{ PersistentSubscription => PS }
+
+import scala.annotation.tailrec
+import scala.collection.immutable.Queue
+
+object PersistentSubscriptionActor {
```

Comment:
I believe I'm already handling this is an integration test - see PersistentSubscriptionActorITest.scala:22, where the connection is reset. Want me to put this in the spec instead?